### PR TITLE
Model updates

### PR DESCRIPTION
**Pull request by torleifg**

### CODE REVIEW

**Comment by asgeirn** [comment]
Have you looked at FINTLabs/fint-betaling-backend#25 ?

**Comment by torleifg** [comment]
> Have you looked at [FINTLabs/fint-betaling-backend#25](https://github.com/FINTLabs/fint-betaling-backend/pull/25) ?

I have included those three models now, and made some adjustments to other parts. Please comment. I'm not sure about Lineitem and OrderItem, maybe they should converge at some point?

**Comment by asgeirn** [comment]
What's the convention for currency amounts?  It seems like I've added `Double` fields, whilst others have `Long` fields?

**Comment by torleifg** [comment]
> What's the convention for currency amounts? It seems like I've added `Double` fields, whilst others have `Long` fields?

According to internet, long or bigdecimal is the preferred choice if calculations are involved...

**Comment by torleifg** [comment]
> In commit that closes #9 we add itemName to OrderItem. But this is not present in the files changed in the pull request. Does this mean we do not need both description and name? Should the description always be the name? This is relevant if we want the product name to be present as a Class in classification-service.
> This update will have conflicts with how the frontend and classification-service work today, hence they will need to be updated to collaborate with the new model.

We can't get both from Fint core, as far as I know, name is mapped to description, others would have to be created in frontend by ex. end-user.

**Comment by OlaStalberg** [comment]
> > In commit that closes #9 we add itemName to OrderItem. But this is not present in the files changed in the pull request. Does this mean we do not need both description and name? Should the description always be the name? This is relevant if we want the product name to be present as a Class in classification-service.
> > This update will have conflicts with how the frontend and classification-service work today, hence they will need to be updated to collaborate with the new model.
> 
> We can't get both from Fint core, as far as I know, name is mapped to description, others would have to be created in frontend by ex. end-user.

Ok. I will use description :)

**Comment by asgeirn** [comment]
> > What's the convention for currency amounts? It seems like I've added `Double` fields, whilst others have `Long` fields?
> 
> According to internet, long or bigdecimal is the preferred choice if calculations are involved...

@OlaStalberg, are you expecting cents and converting in the UI?

**Comment by OlaStalberg** [comment]
> > > What's the convention for currency amounts? It seems like I've added `Double` fields, whilst others have `Long` fields?
> > 
> > 
> > According to internet, long or bigdecimal is the preferred choice if calculations are involved...
> 
> @OlaStalberg, are you expecting cents and converting in the UI?

If I do not misinterpret what you mean: using the existing model convert as seen here:
https://github.com/FINTLabs/fint-betaling-frontend/blob/0ee53bd2d954e711721d455d4a7111aaa8440218/src/feature/payment/step/pick_product/product_table.jsx#L133

File: src/main/java/no/fint/betaling/model/OrderItem.java [hunk 1]
```diff
@@ -5,13 +5,14 @@
 import java.net.URI;
 
 @Data
-public class OrderLine {
-    private URI itemUri;
+public class OrderItem {
+    private String code;
```
Comment (asgeirn) [inline src/main/java/no/fint/betaling/model/OrderItem.java]: Is this the code of `Lineitem` ?  If so, it should perhaps be called `itemCode`.

**Pull request merged**